coopr-docs/docs/source/guide/superadmin/docker-automator-plugin.rst

```diff
@@ -0,0 +1,119 @@
+..
+   Copyright © 2012-2015 Cask Data, Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+ 
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+:orphan:
+
+.. _plugin-reference:
+
+
+.. index::
+   single: Docker Automator Plugin
+
+=======================
+Docker Automator Plugin
+=======================
+
+.. include:: /guide/admin/admin-links.rst
+
+This section describes an automator that uses docker commands via Secure Shell (SSH).
+
+Overview
+========
+
+The Docker Automator plugin, like all automator plugins, is responsible for performing the installation and operation
+of services on remote hosts.  The Docker Automator plugin provides a simple way to install Docker images from Docker Hub
+and to manage containers of these images.
+
+.. note:: The Docker Automator currently requires that Docker is installed, configured, and running on the target host(s), such as provided by CoreOS.
+
+
+As an example, consider the following Coopr service definition:
+::
+
+    {
+        "name": "cdap-standalone",
+        "description": "CDAP SDK (Standalone) running in a Docker container (caskdata/cdap-standalone)",
+        "provisioner": {
+            "actions": {
+                "install": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                },
+                "start": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                },
+                "stop": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                }
+            }
+        }
+    }
+
+This defines a service named "cdap-standalone" which has three defined actions, install, start, and stop.  When each
+action is invoked for this service, the ``type`` field indicates to the provisioner to use the Docker Automator plugin
+to manage the action.  The Docker Automator defines two custom fields: a required ``image_name`` and optional
```

Comment:
Copied from `shell-automator-plugin.rst`
If I change this, we should change it in both places.